core/ratarmountcore/LibArchiveMountSource.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+
+import io
+import os
+import stat
+import time
+from typing import cast, Dict, IO, Iterable, List, Optional, Union
+
+from .MountSource import FileInfo, MountSource
+from .utils import overrides
+
+
+try:
+    import libarchive
+except ImportError:
+    pass
+
+
+class FileInsideArchive(io.RawIOBase):
+    """
+    The seek implementation reopens the file on seeking back.
+    """
+
+    def __init__(self, reopen, file_size):
+        io.RawIOBase.__init__(self)
+        self.reopen = reopen
```

Comment:
```suggestion
        self._reopen = reopen
```

Probably better to make it "private" as it currently isn't used from outside.